x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/RemoteClusterPermissions.java

```diff
@@ -70,19 +79,29 @@ public class RemoteClusterPermissions implements NamedWriteable, ToXContentObjec
     // package private non-final for testing
```

Comment:
> I'd really love to avoid mutable static state

For clarity, this is only mutable as package private for tests and this is fairly common practice.

> if we ever run unit tests in parallel

We never do that in the same JVM. If we did the problems would run much deeper than this concern. 

> then spy ... and mock

I am not a fan of any usage of mocks beyond the absolute simplest case. IMO they end up adding unnecessary complexity to the tests making the difficult to read and understand behavior (which is imo 1/2 the point of the tests) and anything but simple mocks used to help breakup the dependency graph to test variants is IMO a test smell. The alternative I would choose here to address the concern (which I do agree as-is is kinda hacky) would be to use a package private setter gated by a setonce or the like, but that feels unnecessary.